src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionsHandler.cs

```diff
@@ -48,19 +49,51 @@ internal class CodeActionsHandler : ILspServiceDocumentRequestHandler<LSP.CodeAc
             _globalOptions = globalOptions;
         }
 
-        public TextDocumentIdentifier GetTextDocumentIdentifier(CodeActionParams request) => request.TextDocument;
+        public TextDocumentIdentifier GetTextDocumentIdentifier(CodeActionParamsWithOptions request) => request.TextDocument;
 
-        public async Task<LSP.CodeAction[]> HandleRequestAsync(LSP.CodeActionParams request, RequestContext context, CancellationToken cancellationToken)
+        public async Task<LSP.CodeAction[]> HandleRequestAsync(CodeActionParamsWithOptions request, RequestContext context, CancellationToken cancellationToken)
         {
             var document = context.Document;
             Contract.ThrowIfNull(document);
 
-            var options = _globalOptions.GetCodeActionOptionsProvider();
+            var options = new LSPCodeActionOptionsProvider(request.AllowGenerateInHiddenCode, _globalOptions.GetCodeActionOptionsProvider());
```

Comment:
I forget - why do we need a provider here instead of passing in the actual `CodeActionOptions`?  Probably a question for @tmat

Reply:
I was surprised to find it wasn't a `CodeActionOptions` myself, though to be honest I'm not sure I would have known how to override the value if it was, so I wasn't unhappy :)

Reply:
Probably because we call to some code that potentially operates on any document/project in the solution and might need options for a different language than the language of the current document.

Reply:
Would be good to add a test for that scenario.